Signal/src/ViewControllers/ConversationView/Cells/OWSContactShareView.m

```diff
@@ -0,0 +1,174 @@
+    iconCircleView.backgroundColor = [UIColor colorWithRGBHex:0x00ffff];
+    iconCircleView.layer.cornerRadius = self.iconSize * 0.5f;
+    [iconCircleView autoSetDimension:ALDimensionWidth toSize:self.iconSize];
+    [iconCircleView autoSetDimension:ALDimensionHeight toSize:self.iconSize];
+    [iconCircleView setCompressionResistanceHigh];
+    [iconCircleView setContentHuggingHigh];
+
+    // TODO: Use avatar, if present and downloaded. else default.
+    UIImage *image = [UIImage imageNamed:@"attachment_file"];
+    OWSAssert(image);
+    UIImageView *imageView = [UIImageView new];
+    imageView.image = [image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+    imageView.tintColor = self.bubbleBackgroundColor;
+    [iconCircleView addSubview:imageView];
+    [imageView autoCenterInSuperview];
+
+    UILabel *topLabel = [UILabel new];
+    topLabel.text = self.contactShare.displayName;
+    topLabel.textColor = [UIColor blackColor];
+    topLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+    topLabel.font = [UIFont ows_dynamicTypeBodyFont];
+
+    UIStackView *labelsView = [UIStackView new];
+    labelsView.axis = UILayoutConstraintAxisVertical;
+    labelsView.spacing = 2;
+    [labelsView addArrangedSubview:topLabel];
+
+    // TODO: Should we just try to show the _first_ phone number?
+    // What about email?
+    // What if the second phone number is a signal account?
+    NSString *_Nullable firstPhoneNumber = self.contactShare.phoneNumbers.firstObject.phoneNumber;
+    if (firstPhoneNumber.length > 0) {
+        UILabel *bottomLabel = [UILabel new];
+        bottomLabel.text = firstPhoneNumber;
+        // TODO:
+        bottomLabel.textColor = [UIColor ows_darkGrayColor];
+        bottomLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+        bottomLabel.font = [UIFont ows_dynamicTypeCaption1Font];
+        [labelsView addArrangedSubview:bottomLabel];
+    }
+
+    UIImage *disclosureImage =
+        [UIImage imageNamed:(self.isRTL ? @"system_disclosure_indicator_rtl" : @"system_disclosure_indicator")];
+    OWSAssert(disclosureImage);
+    UIImageView *disclosureImageView = [UIImageView new];
+    disclosureImageView.image = [disclosureImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+    disclosureImageView.tintColor = [UIColor blackColor];
+    [disclosureImageView setCompressionResistanceHigh];
+    [disclosureImageView setContentHuggingHigh];
+
+    UIStackView *stackView = [UIStackView new];
+    stackView.axis = UILayoutConstraintAxisHorizontal;
+    stackView.spacing = self.iconHSpacing;
+    stackView.alignment = UIStackViewAlignmentCenter;
+    [contentView addSubview:stackView];
+    [stackView autoPinLeadingToSuperviewMarginWithInset:self.iconHMargin];
+    [stackView autoPinTrailingToSuperviewMarginWithInset:self.iconHMargin];
+    [stackView autoVCenterInSuperview];
+    // NOTE: It's critical that we pin to the superview top and bottom _edge_ and not _margin_.
```

Comment:
In the future can you try to focus comments more on *why* vs. *what* - the code should explain *what* adequately enough.

I don't understand why this is critical - in this case, the stack views superview is a container view, which has 0 margin, so I can't imagine why there'd even be a difference in pinning to edges vs margin.

If this is a no-op, we should get rid of it, IMO superstitious no-op code is just debt that distracts from the actual important logic.

Reply:
I've rewritten these comments.  The intent is for these overflow safe guards to be orthogonal to changes to the margins.